packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol

```diff
@@ -93,57 +113,45 @@ contract LongShortPair is Testable, Lockable {
-     * @param _collateralPerPair how many units of collateral are required to mint one pair of synthetic tokens.
-     * @param _priceIdentifier registered in the DVM for the synthetic.
-     * @param _longToken ERC20 token used as long in the LSP. Mint and burn rights needed by this contract.
-     * @param _shortToken ERC20 token used as short in the LSP. Mint and burn rights needed by this contract.
-     * @param _collateralToken ERC20 token used as collateral in the LSP.
+     * @param params struct to define input parameters to construct the long short pair contract.
      * @param _finder UMA protocol Finder used to discover other protocol contracts.
-     * @param _financialProductLibrary Contract providing settlement payout logic.
-     * @param _customAncillaryData Custom ancillary data to be passed along with the price request. If not needed, this
-     *                             should be left as a 0-length bytes array.
-     * @param _prepaidProposerReward Preloaded reward to incentivize settlement price proposals.
      * @param _timerAddress Contract that stores the current time in a testing environment. Set to 0x0 in production.
      */
     constructor(
-        uint64 _expirationTimestamp,
-        uint256 _collateralPerPair,
-        bytes32 _priceIdentifier,
-        ExpandedIERC20 _longToken,
-        ExpandedIERC20 _shortToken,
-        IERC20 _collateralToken,
+        ConstructorParams memory params,
         FinderInterface _finder,
-        LongShortPairFinancialProductLibrary _financialProductLibrary,
-        bytes memory _customAncillaryData,
-        uint256 _prepaidProposerReward,
         address _timerAddress
     ) Testable(_timerAddress) {
         finder = _finder;
-        require(_expirationTimestamp > getCurrentTime(), "Expiration timestamp in past");
-        require(_collateralPerPair > 0, "Collateral per pair cannot be 0");
-        require(_getIdentifierWhitelist().isIdentifierSupported(_priceIdentifier), "Identifier not registered");
+        require(params.expirationTimestamp > getCurrentTime(), "Expiration timestamp in past");
+        require(params.collateralPerPair > 0, "Collateral per pair cannot be 0");
+        require(_getIdentifierWhitelist().isIdentifierSupported(params.priceIdentifier), "Identifier not registered");
         require(address(_getOptimisticOracle()) != address(0), "Invalid finder");
-        require(address(_financialProductLibrary) != address(0), "Invalid FinancialProductLibrary");
-        require(_getCollateralWhitelist().isOnWhitelist(address(_collateralToken)), "Collateral not whitelisted");
+        require(address(params.financialProductLibrary) != address(0), "Invalid FinancialProductLibrary");
+        require(_getCollateralWhitelist().isOnWhitelist(address(params.collateralToken)), "Collateral not whitelisted");
+        require(params.optimisticOracleLivenessTime > 0, "OO liveness cannot be 0");
```

Comment:
liveness also has to be less than 5200 weeks according to the OptimisticOracle

Reply:
good catch! fixed